doc/design/milestone-0.2.0/csv-generation.md

```diff
@@ -0,0 +1,176 @@
+)
+
+// CSVInstallStrategyUpdate embeds the OLM's install strategy spec.
+type CSVInstallStrategyUpdate struct {
+    *install.StrategyDetailsDeployment
+  
+    // Future utility fields go here.
+}
+
+// getLocalInstallStrategyCache retrieves the local cache singleton and returns the install strategy cache.
+func getLocalInstallStrategyCache() *CSVInstallStrategyUpdate {
+    factory := getLocalCacheFactory()
+    return factory.InstallStrategyCache
+}
+
+// AddDeploymentSpecToCSVInstallStrategyUpdate adds an RBAC Role to the local cache singletons' permissions.
+func AddRoleToCSVInstallStrategyUpdate(yamlDoc []byte) error {
+    localInstallStrategyUpdate := getLocalInstallStrategyCache()
+
+    newRBACRole := new(rbacv1beta1.Role)
+    _ = yaml.Unmarshal(yamlDoc, newRBACRole)
+
+    newPermissions := install.StrategyDeploymentPermissions{
+        ServiceAccountName: newRole.ObjectMeta.Name,
+        Rules:              newRole.Rules,
+    }
+    localInstallStrategyUpdate.Permissions = append(localInstallStrategyUpdate.Permissions, newPermissions)
+
+    return nil
+}
+
+// AddDeploymentSpecToCSVInstallStrategyUpdate adds a Deployment to the local cache singletons' install strategy.
+func AddDeploymentSpecToCSVInstallStrategyUpdate(yamlDoc []byte) error {
+    localInstallStrategyUpdate := getLocalInstallStrategyCache()
+
+    newDeployment := new(appsv1.Deployment)
+    _ = yaml.Unmarshal(yamlDoc, newDeployment)
+  
+    newDeploymentSpec := install.StrategyDeploymentSpec{
+        Name: newDeployment.ObjectMeta.Name,
+        Spec: newDeployment.Spec,
+    }
+    localInstallStrategyUpdate.DeploymentSpecs = append(localInstallStrategyUpdate.DeploymentSpecs, newDeploymentSpec)
+
+    return nil
+}
+
+// Apply applies cached updates in CSVInstallStrategyUpdate to the appropriate csv fields.
+func (us *CSVInstallStrategyUpdate) Apply(csv *v1alpha1.ClusterServiceVersion) error {
+    // Get install strategy from csv.
+    var resolver *install.StrategyResolver
+    strat, _ := resolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+    installStrat, _ := strat.(*install.StrategyDetailsDeployment)
+
+    // Update permissions and deployments with custom field update methods.
+    us.updatePermissions(installStrat)
+    us.updateDeploymentSpecs(installStrat)
+
+    // Re-serialize permissions into csv install strategy.
+    updatedStrat, _ := json.Marshal(installStrat)
+    csv.Spec.InstallStrategy.StrategySpecRaw = updatedStrat
+
+    return nil
+}
+```
+
+### User-defined yaml fields
+
+Many CSV fields cannot be populated using generated, non-SDK-specific manifests. These fields are mostly human-written, English metadata about the operator and various CRD's. Users must directly modify their CSV yaml file, adding personalized data to the following required fields. Users will receive a warning on `operator-sdk build` when a lack of data in any of the required fields is detected.
```

Comment:
Just curious do these match the CSV fields one-to-one? If yes maybe we can add a note to that just for clarity?

Reply:
Yes they do, although there are a few CSV fields that aren't in CRD's. I'll add a note.